src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -119,12 +119,6 @@ public void Flush()
         /// </summary>
         public void Dispose()
         {
-            if (this.snapshotTimer != null)
-            {
-                this.snapshotTimer.Dispose();
-                this.snapshotTimer = null;
-            }
-
             this.Flush();
```

Comment:
We could. Or we could leave it to underline that Flush *must* be called. I think either decision is not a blocker for this change.